content/zh/docs/reference/labels-annotations-taints/_index.md

```diff
@@ -0,0 +1,451 @@
+
+例子：`topology.kubernetes.io/zone=us-east-1c`
+
+用于：Node, PersistentVolume
+
+在 Node 上：`kubelet` 或外部`cloud-controller-manager` 使用 `cloudprovider` 提供的信息填充它。仅当你使用 `cloudprovider` 时才会设置此项。但是，如果它在你的拓扑中有意义，你应该考虑在 Node 上设置它。
+
+在 PersistentVolume 上：拓扑感知卷配置器将自动在 `PersistentVolumes` 上设置 Node 亲和性约束。
+
+一个 Zone 代表一个逻辑故障域。 Kubernetes 集群通常跨越多个 Zone 以提高可用性。虽然 Zone 的确切定义留给基础设施实现，但 Zone 的常见属性包括 Zone 内非常低的网络延迟、 Zone 内的免费网络流量以及与其他 Zone 的故障独立性。例如，一个 Zone 内的 Node 可能共享一个网络交换机，但不同 Zone 中的 Node 不应该。
+
+一个 Region 代表一个更大的域，由一个或多个 Zone 组成。 Kubernetes 集群跨多个 Region 并不常见，虽然 Zone 或 Region 的确切定义留给基础设施实现，但 Region 的共同属性包括它们之间的网络延迟比它们内部更高，它们之间的网络流量成本非零，以及与其他 Zone 或 Region 的故障独立性。例如，一个 Region 内的 Node 可能共享电力基础设施（例如 UPS 或发电机），但不同 Region 的 Node 通常不会。
+
+Kubernetes 对 Zone 和 Region 的结构做了一些假设：
+
+1) Zone 和 Region 是分层的： Zone 是 Region 的严格子集，没有 Zone 可以在两个 Region 中
+
+2) Zone 名称跨 Region 是唯一的；例如， Region “africa-east-1”可能由 Zone “africa-east-1a”和“africa-east-1b”组成
+
+假设拓扑标签不会改变应该是安全的。尽管标签是严格可变的，但它们的消费者可以假设给定 Node 不会在 Zone 之间移动而不会被销毁和重新创建。
+
+Kubernetes 可以通过多种方式使用这些信息。例如，调度程序会自动尝试将 ReplicaSet 中的 Pod 分布在单 Zone 集群中的 Node 之间（为了减少节点故障的影响，请参阅 [kubernetes.io/hostname](#kubernetesiohostname)）。对于多 Zone 集群，这种传播行为也适用于 Zone （以减少 Zone 故障的影响）。这是通过_SelectorSpreadPriority_ 实现的。
+
+_SelectorSpreadPriority_ 是一个尽力而为的布局。如果集群中的 Zone 是异构的（例如：不同数量的节点、不同类型的 Node 或不同的 pod 资源需求），这种布局可能会阻止你的 Pod 跨 Zone 均匀分布。如果需要，你可以使用同质 Zone （相同数量和类型的节点）来减少不均匀分布的可能性。
+
+调度程序（通过 _VolumeZonePredicate_ 谓词）还将确保声明给定卷的 Pod 仅布局在与该卷相同的 Zone 中。卷不能跨 Zone 挂载。
+
+如果 `PersistentVolumeLabel` 不支持你的 PersistentVolumes 的自动标签，你应该考虑手动添加标签（或添加对 `PersistentVolumeLabel` 的支持）。使用 `PersistentVolumeLabel` ，调度程序可以防止 Pod 将卷安装在不同的 Zone 中。如果你的基础架构没有此限制，则根本不需要将 Zone 标签添加到卷中。
+
+### volume.beta.kubernetes.io/storage-provisioner (已弃用)
+
+例子：`volume.beta.kubernetes.io/storage-provisioner: k8s.io/minikube-hostpath`
+
+用于：PersistentVolumeClaim
+
+此注解已被弃用。
+
+### volume.kubernetes.io/storage-provisioner
+
+用于：PersistentVolumeClaim
+
+此注解将添加到动态配置所需的 PVC。
+
+### node.kubernetes.io/windows-build {#nodekubernetesiowindows-build}
+
+例子：`node.kubernetes.io/windows-build=10.0.17763`
+
+用于：Node
+
+当 kubelet 在 Microsoft Windows 上运行时，它会自动标记其 Node 以记录正在使用的 Windows Server 的版本。
+
+标签的值采用“MajorVersion.MinorVersion.BuildNumber”格式。
+
+### service.kubernetes.io/headless {#servicekubernetesioheadless}
+
+例子：`service.kubernetes.io/headless=""`
+
+用于：Service
+
+当拥有的 Service 是无头类型时，控制平面将此标签添加到 Endpoints 对象。
+
+### kubernetes.io/service-name {#kubernetesioservice-name}
+
+例子：`kubernetes.io/service-name="nginx"`
+
+用于：Service
+
+Kubernetes 使用这个标签来区分多个服务。目前仅用于 `ELB` （弹性负载均衡器）。
+
+### endpointslice.kubernetes.io/managed-by {#endpointslicekubernetesiomanaged-by}
+
+例子：`endpointslice.kubernetes.io/managed-by="controller"`
+
+用于：EndpointSlices
+
+标签用于指示管理 EndpointSlice 的控制器或实体。该标签旨在使不同的 EndpointSlice 对象能够由同一集群内的不同控制器或实体管理。
+
+### endpointslice.kubernetes.io/skip-mirror {#endpointslicekubernetesioskip-mirror}
+
+例子：`endpointslice.kubernetes.io/skip-mirror="true"`
+
+用于：Endpoints
```

Comment:
BTW, this API kind is naturally plural (so “Endpoints” is correct).

Reply:
Endpoint maybe best?

Reply:
No. What @sftim meant is that `Endpoints` is the resource name and it IS designed to be plural.